DUI3-DX/Connectors/Autocad/Speckle.Connectors.AutocadShared/HostApp/AutocadIdleManager.cs

```diff
@@ -0,0 +1,38 @@
+using System.Collections.Concurrent;
+
+namespace Speckle.Connectors.Autocad.HostApp;
+
+public class AutocadIdleManager
+{
+  private readonly ConcurrentDictionary<string, Action> _sCalls = new();
+  private bool _hasSubscribed;
+
+  /// <summary>
+  /// Subscribe deferred action to AutocadIdle event to run it whenever Autocad become idle.
+  /// </summary>
+  /// <param name="action"> Action to call whenever Autocad become Idle.</param>
+  public void SubscribeToIdle(Action action)
+  {
+    _sCalls[action.Method.Name] = action;
+
+    if (_hasSubscribed)
+    {
+      return;
+    }
+
+    _hasSubscribed = true;
```

Comment:
this setting and clearing represents a code-level race-condition, but I wonder if it is real world. I think you inherited this pattern from existing DUI3 but we do need to decide whether this is a problem, possibly it is...